### PR TITLE
Fixed #439 - duplicated revision by saving exactly same document

### LINKED ISSUE #439: Jenkins build failure (Unit Test)
http://172.23.113.52:8080/job/couchbase-lite-android-build/40/console

```
com.couchbase.lite.replicator.ReplicationTest > testOneTimePullReplicationSendStoppedOnce[hudson_en-US_160_WVGA_android-19_armeabi-v7a_param(AVD) - 4.4.2] [31mFAILED [0m
	junit.framework.AssertionFailedError: expected:<1> but was:<0>
	at com.couchbase.lite.replicator.ReplicationTest.testOneTimePullReplicationSendStoppedOnce(ReplicationTest.java:4345)
:connectedAndroidTest FAILED

FAILURE: Build failed with an exception.
```

### PR DESCRIPTION
**Pull request by hideki**
- record should be unique by doc_id and redid
- in case of duplicated revision, should return revision instead of null.

### CODE REVIEW

**Comment by hideki** [comment]
Side effects - SQLiteDatabase dumps stacktrace to logcat for conflict. In the future, we should move to `insertOrThrow()` method instead of `insert()` method of `SQLiteDabase`.

**Comment by hideki** [comment]
Original Ticket: https://github.com/couchbase/couchbase-lite-android/issues/439

File: src/main/java/com/couchbase/lite/Database.java [hunk 1]
```diff
@@ -3716,7 +3720,9 @@ public RevisionInternal putRevision(RevisionInternal oldRev, String prevRevId, b
             }
 
             if(newSequence <= 0) {
-                return null; // duplicate rev; see above
+                // duplicate rev; see above
+                notifyChange(newRev, winningRev, null, inConflict);
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/Database.java]: Do we need to set resultStatus to Status.OK here as well?

Comment (hideki) [inline src/main/java/com/couchbase/lite/Database.java]: YES - Thank you for catching this.

**Pull request merged**